src/live_restore/live_restore_fs.c

```diff
@@ -247,17 +265,94 @@ __live_restore_fs_find_layer(WT_FILE_SYSTEM *fs, WT_SESSION_IMPL *session, const
 }
 
 /*
- * __live_restore_fs_notsup --
- *     Return an error message indicating the given functionality is not supported.
+ * __live_restore_fs_directory_list_worker --
+ *     The list is a combination of files from the destination and source directories. For
+ *     destination files, exclude any files matching the marker paths. For source files, exclude
+ *     files that are either marked as tombstones or already present in the destination directory.
  */
 static int
-__live_restore_fs_notsup(WT_SESSION *wt_session)
+__live_restore_fs_directory_list_worker(WT_FILE_SYSTEM *fs, WT_SESSION *wt_session,
+  const char *directory, const char *prefix, char ***dirlistp, uint32_t *countp, bool single)
 {
-    WT_RET_MSG((WT_SESSION_IMPL *)wt_session, ENOTSUP, "Unsupported fs operation");
+    WT_DECL_RET;
```

Comment:
Given we don't have to declare variables at the start this function could possibly be cleaned up a bit, it is likely not a big deal though.